File: Nitya_21GCEBCS097/day2.py
class Solution:

    def sortArrayByParity(self, num: list[int]) -> list[int]:
        j: int = 0

        for i in range(len(num)):

            if num[i] & 1 == 0:

                if i != j:
                    temp = num[i]
                    num[i] = num[j]
                    num[j] = temp

                j += 1

        return num

    def another_solution(self, num: list[int]) -> list[int]:
        j: int = 0

        for i, val in enumerate(num):

            if val & 1 == 0:
                temp = num[i]
                num[i] = num[j]
                num[j] = temp

                j += 1

        return num

File: Shashank Shukla 18-CSE/Day-2.java
//day-2 code two pointers approach 
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int start=0, end = nums.length-1;
        while(start < end){
            if(nums[start] % 2 != 0){
                int temp = nums[start];
                nums[start] = nums[end];
                nums[end] = temp;
                end--;
            } else start++;
        }
        return nums;
    }
}
